### ISSUE TITLE
$queryString issue with "except"

### ISSUE THREAD

**Issue by Dimonka2** [livewire/livewire#1504]
### Description
I have noticed that $queryString with "except" rule does not work as expected. 

I have a several tabs on a page and $activeTab that keeps open tab and this property is added to a query string like this:  

```php
    public $activeTab = 'reviews';

    protected $queryString = [
        'activeTab' => ['except' => 'reviews'],
    ];
```
If I click on other than "reviews" tabs - everything works and query string is updated. If I click on tab "reviews" - tab is shown, but query string keeps last clicked tab (does not remove "activeTab" from the query string). 

The rendered HTML part just in case:

```html
<ul class="nav  mb-0     nav-pills    " role="tablist">
                            <li class="nav-item">
            <a name="reviews" wire:click.prevent="$set("activeTab", "reviews")" data-toggle="tab" aria-selected="true" role="tab" aria-controls="id-100" href="#id-100" class=" nav-link active" active="1"><i class="fa fa-book-reader"></i> Reviews <span class="kt-badge kt-badge--danger">1</span></a>
        </li>
                                    <li class="nav-item">
            <a name="sourceStages" wire:click.prevent="$set("activeTab", "sourceStages")" data-toggle="tab" aria-selected="false" role="tab" aria-controls="id-101" href="#id-101" class=" nav-link"><i class="fa fa-puzzle-piece"></i> Source stages <span class="kt-badge kt-badge--danger">2</span></a>
        </li>
            </ul>

```
**Context**
- Livewire version: 2.0.+
- Laravel version: 7.+
- Browser: Firefox

**Comment by NeptuneDesign** [comment]
Hello I have the same problem.

**Comment by deanzod** [comment]
if that code is literally copied and pasted you have double quotes within double quotes.

```

wire:click.prevent="$set("activeTab", "reviews")"

```

Should be

```


wire:click.prevent="$set('activeTab', 'reviews')"


```

**Comment by nuernbergerA** [comment]
> if that code is literally copied and pasted you have double quotes within double quotes.
> 
> ```
> 
> wire:click.prevent="$set("activeTab", "reviews")"
> 
> ```
> 
> Should be (same for other one too)
> 
> ```
> 
> wire:click.prevent="$set('activeTab', 'reviews')"
> 
> ```
> 
> 

You nailed it

**Comment by Dimonka2** [comment]
> 
> 
> if that code is literally copied and pasted you have double quotes within double quotes.
> 
> ```
> 
> wire:click.prevent="$set("activeTab", "reviews")"
> ```
> 
> Should be (same for other one too)
> 
> ```
> 
> wire:click.prevent="$set('activeTab', 'reviews')"
> ```

This is not an issue. Firefox quoting them right. Everything fires and properties gets updated, but browser query string stays the same ("activeTab=sourceStages") when actual $activeTab changed to "reviews" in the component.

**Comment by Dimonka2** [comment]
Interesting finding. In combination with other active parameters query string gets updated correctly.
It seems that there is a logic like if there is nothing in query string - do not change it (where actually needed to remove previously set value).

**Comment by deanzod** [comment]
I took your code, changed the quotes to single quotes and it works perfectly in LW2, Laravel 7.26.1 + FF with or without other query string parameters. If changing the quotes doesn't work for you there must be an issue in a different part of your code not posted

**Comment by NeptuneDesign** [comment]
The pagination also does not work in the URL there is always displayed page-1

**Comment by deanzod** [comment]
I would guess this is a bug but if you need a quick dirty fix this works:

Add a public variable like

```
public $something;
```

And add that to your $queryString
```
protected $queryString = ['something', 'activeTab' => ['except' => 'reviews']];
```

Not ideal but it will force a `?` on your url on page load which seems to fix the problem

**Comment by calebporzio** [comment]
This should be fixed now in 2.1.0, re-open if not. Thanks!
